### PR TITLE
Reference Update [Batch 1]

### PR DESCRIPTION
**Pull request by nasbench**
This PR aims to 

- Update old references
- Change the GitHub links to permalinks
- Optimize conditions from `s1 and s2` to something like `all of s1`
- Conforme indentation across all rules

This first batch has updated 100+ rules more is to come later.

### CODE REVIEW

**Review by frack113 — COMMENTED** [review]
195 files 😨

File: rules/windows/process_creation/proc_creation_win_non_priv_reg_or_ps.yml [hunk 1]
```diff
@@ -4,43 +4,43 @@ status: test
...
+    registry_folder:
+        CommandLine|contains|all:
+            - 'ControlSet'
+            - 'Services'
+    registry_key:
+        CommandLine|contains:
+            - 'ImagePath'
+            - 'FailureCommand'
+            - 'ServiceDLL'
+    condition: integrity_level and (reg or powershell_1 and powershell_2) and registry_folder and registry_key
```
Comment (frack113) [inline rules/windows/process_creation/proc_creation_win_non_priv_reg_or_ps.yml:36]:
```yaml
detection:
    reg:
        CommandLine|contains|all:
            - 'reg'
            - 'add'
    powershell_1:
        CommandLine|contains: 'powershell'
    powershell_2:
        CommandLine|contains:
            - 'set-itemproperty'
            - ' sp '
            - 'new-itemproperty'
    select_data:
        IntegrityLevel: 'Medium'
        CommandLine|contains|all:
            - 'ControlSet'
            - 'Services'
        CommandLine|contains:
            - 'ImagePath'
            - 'FailureCommand'
            - 'ServiceDLL'
    condition:  (reg or powershell_1 and powershell_2) and select_data
```
:magic_wand:  group the and

File: rules/windows/process_creation/proc_creation_win_susp_print.yml [hunk 2]
```diff
@@ -4,27 +4,27 @@ status: test
...
+    selection1:
+        Image|endswith: \print.exe
+        CommandLine|startswith: print
+    selection2:
+        CommandLine|contains: /D
+    exeCondition:
+        CommandLine|contains: .exe
+    cmdExclude:
+        CommandLine|contains: print.exe
+    condition: selection1 and selection2 and exeCondition and not cmdExclude
```
[the remark this answers is not in the document]
Comment (frack113) [reply rules/windows/process_creation/proc_creation_win_susp_print.yml:24]:
```yaml
   selection:
        Image|endswith: \print.exe
        CommandLine|startswith: print
        CommandLine|contains|all:
           - /D
          - .exe
    filter_print:
        CommandLine|contains: print.exe
    condition: selection and not filter_print
```
:laughing:

**Review by frack113 — APPROVED** [review]

**Comment by Neo23x0** [comment]
Nice!

**Pull request merged**